FoodTracker/Controllers/MealTableViewController.swift

```diff
@@ -0,0 +1,163 @@
+//
+//  MealTableViewController.swift
+//  FoodTracker
+//
+//  Created by BrainX IOS 4 on 2020-12-14.
+//
+
+import UIKit
+import os.log
+
+class MealTableViewController: UIViewController {
+    
+    //MARK: -Properties
+    @IBOutlet var tableView: UITableView!
+    var meals = [Meal]()
+    
+    
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView.delegate = self
+        
+        navigationItem.leftBarButtonItem = editButtonItem
+
+        loadData()
+    }
+
+    
+    // MARK: - Navigation Methods
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        
+        super.prepare(for: segue, sender: sender)
+        
+        switch(segue.identifier){
+        case "AddItem":
+            os_log("Adding a new meal.", log: OSLog.default, type: .debug)
+        case "showDetail":
+            guard let mealDetailViewController = segue.destination as? MealViewController else{
+                fatalError("Unexpected Controller: \(segue.destination)")
+            }
+            guard let mealTableViewCell = sender as? MealTableViewCell else{
+                fatalError("Unexpected Cell: \(sender ?? "")")
+            }
+            guard let indexPath = tableView.indexPath(for: mealTableViewCell) else{
+                fatalError("Selected cell is not in table.")
+            }
+            
+            let selectedMeal = meals[indexPath.row]
+            mealDetailViewController.meal = selectedMeal
+        default:
+            fatalError("Unexpected Segue Identifier; \(segue.identifier ?? "")")
+        }
+    }
+    
+    @IBAction func unwindToMealList(sender: UIStoryboardSegue){
```

Comment:
All the methods that be called on user action we write them under the `Actions` mark.
And add a line break after the @objc and @IBAction when used with methods.

Reply:
Done